### PR TITLE
Cleanup mypy errors

### PR DESCRIPTION
**Pull request by Obs01ete**
## Description

Cleaning up mypy errors.

## Motivation and Context

Partially solves the issue #159

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] Wave 1
- [ ] Wave 2

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/agents/role_playing.py [hunk 1]
```diff
@@ -266,17 +266,15 @@ def step(
...
         if user_response.terminated or user_response.msgs is None:
             return (ChatAgentResponse([], False, {}),
                     ChatAgentResponse([], user_response.terminated,
                                       user_response.info))
         user_msg = self.process_messages(user_response.msgs)
         self.user_agent.update_messages(user_msg)
 
-        assistant_response = self.assistant_agent.step(
-            user_msg.to_user_chat_message())
+        assistant_response = self.assistant_agent.step(user_msg)
```
Comment (lightaime) [inline camel/agents/role_playing.py:277]: How does this work as expected? We need to set `role` to `'user'` right?

Comment (Obs01ete) [reply camel/agents/role_playing.py:277]: user_msg is UserChatMessage and step takes ChatMessage as input. A derived class can be always cast to the base class. OOO baby!

Comment (Obs01ete) [reply camel/agents/role_playing.py:277]: @lightaime what was the original purpose of
```
self.assistant_agent.step(
            user_msg.to_user_chat_message())
```
Convert a user message to a user massage?... user_msg is a CatMessage anyway

File: camel/agents/role_playing.py [hunk 2]
```diff
@@ -266,17 +266,17 @@ def step(
             whether or not the user agent terminated the conversation, and
             any additional user information.
         """
-        user_response = self.user_agent.step(
-            assistant_msg.to_user_chat_message())
+        assistant_msg_rst = assistant_msg.set_user_role_at_backend()
```
Comment (lightaime) [inline camel/agents/role_playing.py:269]: What does `rst` stand for?

File: camel/generators.py [hunk 3]
```diff
@@ -150,19 +152,19 @@ def __init__(self, assistant_role_names_path:
 
         if assistant_role_names is None:
             with open(assistant_role_names_path, "r") as f:
-                assistant_role_names: List[str] = f.read().splitlines()
+                assistant_role_names_: List[str] = f.read().splitlines()
```
Comment (lightaime) [inline camel/generators.py:155]: Why do we need this underscore?

**Review by lightaime — APPROVED** [review]
Thanks! LGTM!

File: camel/agents/role_playing.py (hunk 2 above)
Comment (Obs01ete) [reply camel/agents/role_playing.py:269]: It was "reset" before. Can we leave it as is?

File: camel/generators.py (hunk 3 above)
Comment (Obs01ete) [reply camel/generators.py:155]: Otherwise mypy complains that assistant_role_names shadows another assistant_role_names from earlier on in the function.

Comment (Obs01ete) [reply camel/generators.py:155]: I could not come up with a good suffix :)

File: camel/agents/role_playing.py (hunk 2 above)
Comment (lightaime) [reply camel/agents/role_playing.py:269]: Sure

**Pull request merged**